src/anonymousUserTracking/criteriaCompletionChecker.ts

```diff
@@ -439,6 +470,16 @@ class CriteriaCompletionChecker {
     }
   }
 
+  private issetCheck(matchObj: any): boolean {
```

Comment:
If this is a json obj, 

```suggestion
  private issetCheck(matchObj: Record<any,any>): boolean {
```

Reply:
not a json object.

Reply:
then what is it? would prefer to be explicit here

Reply:
please properly type this. after that is done, can merge.